api/app/GraphQL/Mutations/SyncPoolScreeningQuestions.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace App\GraphQL\Mutations;
+
+use App\Enums\AssessmentStepType;
+use App\Models\AssessmentStep;
+use App\Models\Pool;
+use App\Models\PoolSkill;
+use App\Models\ScreeningQuestion;
+use Exception;
+use Illuminate\Support\Facades\DB;
+
+final class SyncPoolScreeningQuestions
+{
+    /**
+     * @param  null  $_
+     * @param  array{}  $args
+     */
+    public function __invoke($_, array $args)
+    {
+        // Run everything in a transaction
+        DB::beginTransaction();
+        try {
+            $pool = Pool::find($args['poolId']);
+            $existingQuestions = ScreeningQuestion::where('pool_id', '=', $args['poolId'])->get();
+            $incomingQuestions = is_array($args['screeningQuestions']) ? $args['screeningQuestions'] : [];
+            $incomingAssessmentStep = $args['assessmentStep'];
+            $incomingQuestionIds = [];
+
+            // Create/update incoming questions based on the existence of an id
+            foreach ($incomingQuestions as $incomingQuestion) {
+                if (isset($incomingQuestion['id'])) {
+                    array_push($incomingQuestionIds, $incomingQuestion['id']);
+                    $questionToUpdate = $existingQuestions->find($incomingQuestion['id']);
+                    if ($questionToUpdate === null) {
+                        throw new Exception('Screening question with id "'.$incomingQuestion['id'].'" does not exist.');
```

Comment:
Any i18n strings should be moved to `packages/i18n/src/messages/apiMessages.ts` so that they can be translated.